frameworks/keyed/react-hookstate/src/main.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { useStateLink, createStateLink, None, Downgraded } from '@hookstate/core';
+
+function random(max) { return Math.round(Math.random() * 1000) % max; }
+
+const A = ["pretty", "large", "big", "small", "tall", "short", "long", "handsome", "plain", "quaint", "clean",
+    "elegant", "easy", "angry", "crazy", "helpful", "mushy", "odd", "unsightly", "adorable", "important", "inexpensive",
+    "cheap", "expensive", "fancy"];
+const C = ["red", "yellow", "blue", "green", "pink", "brown", "purple", "brown", "white", "black", "orange"];
+const N = ["table", "chair", "house", "bbq", "desk", "car", "pony", "cookie", "sandwich", "burger", "pizza", "mouse",
+    "keyboard"];
+
+let nextId = 1;
+
+function buildData(count) {
+    const data = {};
+    for (let i = 0; i < count; i++) {
+        data[nextId] = {
+            id: nextId,
+            label: `${A[random(A.length)]} ${C[random(C.length)]} ${N[random(N.length)]}`,
+        };
+        nextId += 1;
+    }
+    return data;
+}
+
+const globalState = createStateLink({});
+let selectedState = undefined;
```

Comment:
Wouldn't it look nice if we put `selectedState` in `globalState`?
I mean I think the pro of your lib is "immutable updates" (correct me if I'm wrong), and having mutable state here seems unfortunate. I assume the change wouldn't affect the performance.

Reply:
The stat can be mutated only via StateLink.set() API. Not sure if it satisfies the criteria of immutability. Regarding the selected state. It is actually within the global state - each row in the state has got field `selected`. This variable is a fast index variable / cache, which points to the StateLink from globalState, which had `selected` field set to true last time.

Reply:
> immutability

I might have misunderstood from your original implementation. So, your lib is more like MobX. (Which implies React.memo won't work as expected in some cases. Sorry, this discussion is nothing to do with benchmark. Let's do it in the other thread.)

> fast index variable / cache

That sounds unrealistic in the typical React coding pattern. You might want at least to put it in Context/useRef. The benchmark shouldn't be too optimized.

Reply:
I had it in the useRef, makes no difference for performance. But if you have got strong feelings about it, I can move it back to useRef. Alternatively, I can do search for unselected unknown selected. It should not change much too. I will keep it as it is, unless you speak up.

Reply:
> I had it in the useRef, makes no difference for performance.

Yeah, that was my expectation.
So, in your real app, do you use `useRef` for the cache?

> I can do search for unselecting unknown selected.

You mean without `let selectedState` or `useRef`? I'm interested in it.